config/locales/blacklight.en.yml

```diff
@@ -140,7 +140,7 @@ en:
 
     search:
       title: '%{application_name} Search Results'
-      search_results_header: 'Search'
+      search_results_header: 'Search Constraints'
```

Comment:
Happy to take suggestions here. I think that the text should identify what is placed in the section. Using Blacklight terminology "Search Constraints" was the best thing that @jvine and I came up with.